lib/buffer.js

```diff
@@ -264,8 +264,8 @@ function fromObject(obj) {
   }
 
   if (obj) {
-    if (isArrayBuffer(obj.buffer) || 'length' in obj ||
-        isSharedArrayBuffer(obj)) {
+    if ('length' in obj || isArrayBuffer(obj.buffer) ||
```

Comment:
That would break hypothetical code which actually explicitly sets `obj.length = undefined`, right? I would be okay with that, just making sure.